routing/payment_lifecycle.go

```diff
@@ -1059,3 +1020,97 @@ func (p *paymentLifecycle) refreshPayment() (DBMPPayment,
 
 	return payment, ps, nil
 }
+
+// processSwitchResults reads the `p.results` map and process the results
+// returned from the htlcswitch.
+func (p *paymentLifecycle) processSwitchResults() error {
+	// Create a slice to remember the results of the attempts that we have
+	// processed.
+	attempts := make([]*channeldb.HTLCAttempt, 0, p.switchResults.Len())
+
+	var errReturned error
+
+	// Range over the map to process all the results.
+	p.switchResults.Range(func(a *channeldb.HTLCAttempt,
+		result *htlcswitch.PaymentResult) bool {
+
+		// Save the keys so we know which items to delete from the map.
+		attempts = append(attempts, a)
```

Comment:
yeah I thought about this too - since the only error we get here is DB-related, I don't think retrying it again would help. This is also the current behavior. But I guess some logs would help so will add!